### DISCUSSION TITLE
Results on stationary MRMS-only data

### DISCUSSION THREAD

**Discussion by ValterFallenius** [openclimatefix/metnet#29] [Show and tell]
I have finalized my model with a simpler setup than I initially planned. With 8 leadtimes and 15 minute spacing the model achieves something but I after rigorous testing the results are still pretty poor. See F1-score plotted below when compared with persistence:

![f1score 0 2 mm](https://user-images.githubusercontent.com/21970939/164726575-d90a18c0-c1f4-4d2a-9121-9f9468b04532.PNG)

I suspect I have too few training samples. MetNet uses 1.7M data samples before they stopped observing overfitting. I have trained a network on 4,400 samples that does something but it doesn't perform nearly as well as the MetNet. In the beginning of my project I decided to make it easy for myself and work with a stationary model, this lead to less data available. 

My available data before preprocessing: 5 years, 365 days, 90 minute data samples ---> 30,000 samples

After sorting out all samples with less than 5 pixels of rain in any lead time only 4,400 samples remain.

MetNet has an input patch of size 1024km x 1024km and a total coverage of 7000km x 2500km, this gives ~15 non-overlapping geographical locations.

MetNets available data: 1.5 years, 365 days, 90 minute samples, 15 geographical non-overlapping locations ---> 131,000 samples

Same sorting technique leaves only ~20,000 samples. 

Since this is way less than 1.7M we can assume they do not use non-overlapping geographical locations, instead this is randomly sampled with yields many more data points.

Some examples of successes and fails:
![successful exampl](https://user-images.githubusercontent.com/21970939/164731274-132785fb-175b-4074-98e1-99e791deb271.PNG)
![successful example 2](https://user-images.githubusercontent.com/21970939/164731294-99b8ef23-ba29-4011-96b7-f26c3381f561.PNG)
![failed example 2](https://user-images.githubusercontent.com/21970939/164731325-8af7b451-454a-4997-ae12-f2380899afca.PNG)
![failed example](https://user-images.githubusercontent.com/21970939/164731310-372425d9-e6ab-4dbe-a65d-bb5ca811358b.PNG)

I am contemplating implementing a non-stationary model, however this would require some time that I might not have since my thesis is due in 1 month.

**Comment by JackKelly** [comment]
Thanks loads for sharing!

Another option might be to pre-train MetNet on a different dataset (e.g. the same dataset that the MetNet authors used) and then fine-tune on your dataset. I think @jacobbieker is busy uploading at least some of the relevant data.

Another option might be to use a simpler model? We were recently involved in an ML competition where the task was to predict the next 2 hours of satellite imagery. U-Nets did really well. Here's the winning ML model: https://github.com/jmather625/climatehack

**Reply by jacobbieker** [reply]
Yeah, I'm working on adding an expanded copy of the MetNet and MetNet-2 data. Currently, only have US MRMS data available though here: https://huggingface.co/datasets/openclimatefix/mrms You can just download the data files individually, as the dataset script isn't ready yet. My computer is currently uploading the full 2019, and most of 2018/2016 data now, and already there is the full 2017,2021, and first few months of 2022 uploaded. So at the least, it should allow for right now 2.25ish years of data, and soon 4-4.5 years of data I think either later today or early next week.

Each year is the 2-minutely PrecipRate output from MRMS for the whole continental US, so roughly 250,000 measurements for each year.

**Reply by jacobbieker** [reply]
I have metadata with kerchunk for streaming in data from GOES-16/17 but there will be more work before I can get it to work correctly and upload to HF, so it might or might not be ready in time for your thesis, unfortunately.

**Reply by ValterFallenius** [reply]
Hello again,

I have implemented a network where the frame is moving, providing me with 16 times more data (non-overlapping targets) on the Swedish dataset. I probably won't be using this for my thesis since the initial results don't seem to improve that much, but it will be interesting to see what it does.

This is not so important but as I am writing my thesis I wanted to cite this repo and I just found out it's possible to implement automatic citation generator in GitHub by adding a CITATION.cff file to a [repo](https://academia.stackexchange.com/questions/14010/how-do-you-cite-a-github-repository). I already did a manual bibtex citation but it could be a good idea for you, especially so credit is given where credit is due, are there more people working on MetNet than you two @JackKelly @jacobbieker ?

/Valter
